### PR TITLE
[Customer Case] Adding test for syncing inventory with missing ip

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Customer Case automation for SAT-24805. Sync inventory status with missing IP

### CODE REVIEW

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_inventory.py -k 'test_positive_sync_inventory_status_missing_host_ip'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8863
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_rhcloud_inventory.py -k test_positive_sync_inventory_status_missing_host_ip --external-logging
Test Result : ======== 6 passed, 22 deselected, 2468 warnings in 10319.90s (2:51:59) =========
```

File: tests/foreman/cli/test_rhcloud_inventory.py [hunk 1]
```diff
@@ -190,6 +190,75 @@ def test_positive_sync_inventory_status(
...
+    )
+    host_status = None
+    for task in task_output:
+        if task.input.get("organization_ids") is None:
+            continue
+        if str(task.input.get("organization_ids")[0]) == str(org.id):
+            host_status = task.output
+            break
+    assert host_status['host_statuses']['sync'] == 2
+    assert host_status['host_statuses']['disconnect'] == 0
```
Comment (vsedmik) [inline tests/foreman/cli/test_rhcloud_inventory.py:259]: Some context on these `2` and `0` values might be helpful in the future for anyone who looks here.

**Review by vsedmik — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_rhcloud_inventory.py [hunk 2]
```diff
@@ -190,6 +190,75 @@ def test_positive_sync_inventory_status(
...
+    task_output = module_target_sat.api.ForemanTask().search(
+        query={'search': f'{inventory_sync_task} and started_at >= "{timestamp}"'}
+    )
+    host_status = None
+    for task in task_output:
+        if task.input.get("organization_ids") is None:
+            continue
+        if str(task.input.get("organization_ids")[0]) == str(org.id):
+            host_status = task.output
+            break
```
[the remark this answers is not in the document]
Comment (ColeHiggins2) [reply tests/foreman/cli/test_rhcloud_inventory.py:257]: I tried that. But could not get it to work... even on the UI itself

File: tests/foreman/cli/test_rhcloud_inventory.py (hunk 1 above)
Comment (ColeHiggins2) [reply tests/foreman/cli/test_rhcloud_inventory.py:259]: Pulled from the  rhcloud_registered_hosts fixture. Two hosts are spun up and registered with insights. In our case, the first one has a missing IP. Both however should be registered and synced to rh cloud without failure.

**Review by sambible — APPROVED** [review]

**Pull request merged**